src/components/application_manager/src/application_data_impl.cc

```diff
@@ -616,6 +626,33 @@ void DynamicApplicationDataImpl::set_reset_global_properties_active(
   is_reset_global_properties_active_ = active;
 }
 
+void DynamicApplicationDataImpl::set_choice_set_allow_mode(
+    const uint32_t choice_set_id, const bool is_allowed) {
+  auto choice_set = choice_set_allowed_map_.find(choice_set_id);
+  if (choice_set_allowed_map_.end() == choice_set) {
+    LOG4CXX_WARN(logger_,
+                 "Choice set with id " << choice_set_id << " is not found.");
+    return;
+  }
+  choice_set->second = is_allowed;
+  LOG4CXX_DEBUG(logger_,
+                "choice_set_id: "
+                    << choice_set_id
+                    << (choice_set->second ? " is allowed" : " disallowed"));
+}
+
+bool DynamicApplicationDataImpl::is_choice_set_allowed(
+    const uint32_t choice_set_id) const {
+  LOG4CXX_DEBUG(logger_, "Choice setID: " << choice_set_id);
+  auto it = choice_set_allowed_map_.find(choice_set_id);
```

Comment:
@IGapchuk `const` ?

Reply:
@ZhdanovP @ZhdanovP Fixed in c03d2e5028617d584edb7e8ca84aa2fc3b1139f2.